web-app/HeimScripts/heatmap/run.R

```diff
@@ -95,6 +154,40 @@ addStats <- function(df, sorting) {
   return(df)
 }
 
+# Coefficient of variation
+coVar     <- function(x) ( sd(x)/mean(x) )
+
+# Specific implementation of range.
+normRange <- function(x) {
+  x <- removeOutliers(x)
+  max(x) - min(x)
+}
+
+# We define outliers as measurements falling outside .25 or .75
+# quantiles by more than 1.5 of interquantile range.
+removeOutliers <- function(x) {
+  qnt    <- quantile(x, probs=c(.25, .75), na.rm = TRUE)
+  H      <- 1.5 * IQR(x, na.rm = TRUE)
+  result <- x[!(x < (qnt[1] - H))]  # Below .25 by more than 1.5 IQR
+  result[!(result > (qnt[2] + H))]  # Above .75 by more than 1.5 IQR
+}
+
+getRankingMethod <- function(rankingMethodName) {
+  if (rankingMethodName == "variance") {
+    return(var)
+  } else if (rankingMethodName == "coef") {
+    return(coVar)
+  } else if (rankingMethodName == "range") {
+    return(normRange)
+  } else if (rankingMethodName == "mean") {
+    return(mean)
+  } else if (rankingMethodName == "median") {
+    return(median)
+  } else {
```

Comment:
is not super clean: if there's invalid input for 'ranking' argument to main() it will return "limma" as rankingMethod. However this will later result in an error when 'ranking' is used to select the appropriate bit of the output of the lmFit method as this will not match any of the names in the output of lmFit.